### PR TITLE
chore: migrate to markdownlint-cli2

### PR DESCRIPTION
**Pull request by nschonni**
Now that we have a custom rule that is used, it's helpful to migrate to the new config that allows declaring the rules in the config.
I added some comments around the rules, since this file allows that.
This config is compatible with the VScod plug-in, so you get the custom rule errors in the IDE

### CODE REVIEW

File: .markdownlint-cli2.jsonc [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+      "style": "fenced"
+    },
+    "MD049": {
+      "style": "underscore"
+    },
+    "MD050": {
+      "style": "asterisk"
+    },
+    // Yari uses underscores rather than dash, and there is now config to use and alternate
+    // algorythim currently
```
Comment (MrBrain295) [inline .markdownlint-cli2.jsonc:116]:
```suggestion
    // algorithm currently
```

File: .markdownlint-cli2.jsonc [hunk 2]
```diff
@@ -0,0 +1,164 @@
+{
```
Comment (caugner) [inline .markdownlint-cli2.jsonc:1]: Does this warrant updating the JSON check workflow (if prettier supports `*.jsonc`)?

https://github.com/mdn/content/blob/cb22605df3ea99dda17c1dc6fcee2ed5b687e4a5/.github/workflows/pr-check_json.yml#L7-L8

File: .markdownlint-cli2.jsonc [hunk 3]
```diff
@@ -0,0 +1,164 @@
...
+  "config": {
+    "default": true,
+    // Some Callouts currently include headings that don't follow this rule
```
Comment (caugner) [inline .markdownlint-cli2.jsonc:4]: As the rule names are not self-explanatory, I would suggest repeating their description here:

```suggestion
    // MD001 - Heading levels should only increment by one level at a time.
    // -> Disabled, as some callouts include headings.
```

File: .markdownlint-cli2.jsonc [hunk 4]
```diff
@@ -0,0 +1,164 @@
...
+      "style": "fenced"
+    },
+    "MD049": {
+      "style": "underscore"
+    },
+    "MD050": {
+      "style": "asterisk"
+    },
+    // Yari uses underscores rather than dash, and there is now config to use and alternate
+    // algorithm currently
```
Comment (caugner) [inline .markdownlint-cli2.jsonc:114]:
```suggestion
    // MD051 - Link fragments should be valid.
    // -> Disabled, as yari generates link fragments by replacing spaces with underscores, not dashes.
```

**Review by caugner — CHANGES_REQUESTED** [review]
Switching from markdownlint-cli to markdownlint-cli2 looks good to me.

**Review by caugner — APPROVED** [review]

**Comment by nschonni** [comment]
FYI @OnkarRuikar @teoli2003

**Comment by Josh-Cena** [comment]
@nschonni After pulling the new changes, my VS Code keeps displaying this at the bottom, and I can neither dismiss the entire panel nor switch to "terminal":

image

Could you investigate? Thanks!

**Comment by nschonni** [comment]
have you run Yarn recently to pull in the custom rule?

**Comment by Josh-Cena** [comment]
Yes, I did. It's installed, but locally. The VS Code extension is only searching in its own bundle, though. I found a setting called `markdownlint.customRules`, but I can't spend too much time on this before going to bed. It's only coming up after this PR.

**Comment by nschonni** [comment]
Ah, I haven't tried this on my Mac since this landed. I'll check later and see if I get the same. 
Quick hack would be to comment out that rule in the `.markdownlint-cli2.jsonc` temporarily.
Adding the settings and suggested extensions probably is a good discussion to spin up

**Comment by teoli2003** [comment]
We want to add settings and suggested extensions on a meta-docs page. I would be glad to see this going forward.

**Comment by nschonni** [comment]
Just tried it on my Mac and there is no issue. I'm running

Version: 1.71.0
Commit: 784b0177c56c607789f9638da7b6bf3230d47a8c
Date: 2022-09-01T07:25:38.437Z
Electron: 19.0.12
Chromium: 102.0.5005.167
Node.js: 16.14.2
V8: 10.2.154.15-electron.0
OS: Darwin x64 19.6.0
Sandboxed: No

**Comment by nschonni** [comment]
I had some user level customRules for the Microsoft Docs Authoring Pack, but I cleared them out to see if it made any difference

**Comment by OnkarRuikar** [comment]
> We want to add settings and suggested extensions on a meta-docs page. I would be glad to see this going forward.

We definitely need to suggest the [markdownlint VSCode plugin](https://marketplace.visualstudio.com/items?itemName=DavidAnson.vscode-markdownlint):

vscodePlugin

I ran `yarn` on the latest code. Then after installing the plugin it worked out of the box. It uses the same `.markdownlint-cli2.jsonc` config file that we added for the cli2 tool.
Note it's flagging our custom rules and the `Quick Fix...` option also works! :+1:  

***
Also, the screenshot shows `yarn lint:md` worked without any error. I am using Linux.

**Comment by Josh-Cena** [comment]
Ah, it only works for me after reloading my VS Code. I guess I was too tired last night. Thanks y'all!

**Pull request merged**